AppOptions: Define Outdir in the code raise a synth ERROR

### Expected Behavior

With python, I want define the output dir directly in the code (without specify it in the command line) when I create the App object like : 
```python
#!/usr/bin/env python
from cdktf import App

cdktf_environment = App(outdir='out-environment-xxxxxxx.out')

# Custom CDKTF code ....

cdktf_environment.synth()
```
Synth cmd is : 
```bash
cdktf synth -a "pipenv run python ./environment-xxxxxxx.py" 
```


### Actual Behavior

When we execute the synth, we get the following error (error code 1): 
```bash
cdktf synth -a "pipenv run python ./environment-xxxxxxx.py" 

[2024-01-10T15:06:22.324] [ERROR] default - ERROR: synthesis failed, because app was expected to call 'synth()', but didn't. Thus "cdktf.out/manifest.json"  was not created: Error: Could not find manifest file at cdktf.out/manifest.json. In case --skip-synth was passed, please try again without the flag.   
```

The output itself is well generated, we could go in the stack generated file and execute terraform init/apply without any issue.

If we specify the ouput dir in the command line with `-o`, the issue don't happen

### Steps to Reproduce

1. Setup your cdktf with python project
2. Define the App object with the `output` param like `App(outdir='out-environment-xxxxxxx.out')`
3. Try to synth without the `-o` param

### Versions

**language**: Python 3.10.13 
**cdktf-cli**: 0.20.0
**node**: v20.9.0
**Terraform**: v1.6.6
**OS**: Linux x86_64
**Python dependency:** 
- cdktf = "~=0.20"
- constructs = "*"

### Providers

Python dependency: 
```
	cdktf-cdktf-provider-kubernetes = "~=11.0"
	cdktf-cdktf-provider-helm = "~=10.0"
	cdktf-cdktf-provider-postgresql = "~=11.0"
	cdktf-cdktf-provider-azurerm = "~=12.0"
	cdktf-cdktf-provider-azuread = "~=12.0"
	cdktf-cdktf-provider-random = "~=11.0"
	cdktf-cdktf-provider-time = "~=10.0"
```
### Gist

_No response_

### Possible Solutions

_No response_

### Workarounds

_No response_

### Anything Else?

_No response_

### References

https://github.com/hashicorp/terraform-cdk/pull/3334

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/packages/cdktf/lib/app.ts
+++ b/packages/cdktf/lib/app.ts
@@ -15,6 +15,15 @@ export interface AppConfig {
   /**
    * The directory to output Terraform resources.
    *
+   * If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+   * - The `-o` / `--output` CLI option
+   * - The `CDKTF_OUTDIR` environment variable
+   * - The `outdir` key in `cdktf.json`
+   *
+   * If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+   *
+   * The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+   *
    * @default - CDKTF_OUTDIR if defined, otherwise "cdktf.out"
    */
   readonly outdir?: string;

--- a/website/docs/cdktf/api-reference/csharp.mdx
+++ b/website/docs/cdktf/api-reference/csharp.mdx
@@ -12442,6 +12442,16 @@ public string Outdir { get; set; }
 
 The directory to output Terraform resources.
 
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+
 ---
 
 ##### `SkipBackendValidation`<sup>Optional</sup> <a name="SkipBackendValidation" id="cdktf.AppConfig.property.skipBackendValidation"></a>

--- a/website/docs/cdktf/api-reference/go.mdx
+++ b/website/docs/cdktf/api-reference/go.mdx
@@ -12442,6 +12442,16 @@ Outdir *string
 
 The directory to output Terraform resources.
 
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+
 ---
 
 ##### `SkipBackendValidation`<sup>Optional</sup> <a name="SkipBackendValidation" id="cdktf.AppConfig.property.skipBackendValidation"></a>

--- a/website/docs/cdktf/api-reference/java.mdx
+++ b/website/docs/cdktf/api-reference/java.mdx
@@ -66,6 +66,16 @@ Context can be read from any construct using `node.getContext(key)`.
 
 The directory to output Terraform resources.
 
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+
 ---
 
 ##### `skipBackendValidation`<sup>Optional</sup> <a name="skipBackendValidation" id="cdktf.App.Initializer.parameter.skipBackendValidation"></a>
@@ -17242,6 +17252,16 @@ public java.lang.String getOutdir();
 - _Default:_ CDKTF_OUTDIR if defined, otherwise "cdktf.out"
 
 The directory to output Terraform resources.
+
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
 
 ---
 

--- a/website/docs/cdktf/api-reference/python.mdx
+++ b/website/docs/cdktf/api-reference/python.mdx
@@ -66,6 +66,16 @@ Context can be read from any construct using `node.getContext(key)`.
 
 The directory to output Terraform resources.
 
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+
 ---
 
 ##### `skip_backend_validation`<sup>Optional</sup> <a name="skip_backend_validation" id="cdktf.App.Initializer.parameter.skipBackendValidation"></a>
@@ -17990,6 +18000,16 @@ outdir: str
 - _Default:_ CDKTF_OUTDIR if defined, otherwise "cdktf.out"
 
 The directory to output Terraform resources.
+
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
 
 ---
 

--- a/website/docs/cdktf/api-reference/typescript.mdx
+++ b/website/docs/cdktf/api-reference/typescript.mdx
@@ -12435,6 +12435,16 @@ public readonly outdir: string;
 
 The directory to output Terraform resources.
 
+If you are using the CDKTF CLI, this value is automatically set from one of the following three sources:
+
+- The `-o` / `--output` CLI option
+- The `CDKTF_OUTDIR` environment variable
+- The `outdir` key in `cdktf.json`
+
+If you are using the CDKTF CLI and want to set a different value here, you will also need to set the same value via one of the three ways specified above.
+
+The most common case to set this value is when you are using the CDKTF library directly (e.g. when writing unit tests).
+
 ---
 
 ##### `skipBackendValidation`<sup>Optional</sup> <a name="skipBackendValidation" id="cdktf.AppConfig.property.skipBackendValidation"></a>

